capa/render/result_document.py

```diff
@@ -183,6 +183,12 @@ class StatementNode(FrozenModel):
     type = "statement"
     statement: Statement
 
+    class Config:
+        # Note! we incur a performance penalty using smart_union because Pydantic checks
+        # all allowed types before even trying to coerce
+        # see https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union
+        smart_union = True
```

Comment:
I still don't quite see why we need `smart_union` here, since all the union candidates are distinct and none are "more specific" than another. That being said, if it doesn't work without `smart_union` and none of us want to become pydantic ninjas, keep it and move along :-)

Reply:
The newest commit no longer uses smart unions - I believe you may have been looking at an older commit here.